Source/JavaScriptCore/wasm/WasmOperations.cpp

```diff
@@ -1119,28 +1126,56 @@ JSC_DEFINE_JIT_OPERATION(operationWasmArrayNew, EncodedJSValue, (Instance* insta
     ASSERT(arraySignature.is<ArrayType>());
     Wasm::FieldType fieldType = arraySignature.as<ArrayType>()->elementType();
 
+    size_t elementSize = fieldType.type.elementSize();
+
     JSWebAssemblyArray* array = nullptr;
-    switch (fieldType.type.kind) {
-    case Wasm::TypeKind::I32:
-    case Wasm::TypeKind::F32: {
+
+    switch (elementSize) {
+    case sizeof(uint8_t): {
+        FixedVector<uint8_t> values(size);
+        for (unsigned i = 0; i < size; i++) {
+            // `encValue` must be an unboxed int (since the typechecker guarantees that its type is i32);
```

Comment:
Let's add an assert here just in case, to aid the fuzzer

Reply:
Done, added a `ASSERT(encValue <= UINT32_MAX);` here